src/backend/distributed/executor/insert_select_executor.c

```diff
@@ -885,11 +889,13 @@ CastExpr(Expr *expr, Oid sourceType, Oid targetType, Oid targetCollation,
 bool
 IsSupportedRedistributionTarget(Oid targetRelationId)
 {
-	CitusTableCacheEntry *tableEntry = GetCitusTableCacheEntry(targetRelationId);
+	CitusTableCacheEntryRef *tableRef = GetCitusTableCacheEntry(targetRelationId);
```

Comment:
We should more aggressively use `PartitionMethod` helper function. Those helper functions should also be consolidated into `metadata_cache.c`